crates/uv-trampoline/src/bounce.rs

```diff
@@ -345,7 +360,82 @@ pub fn bounce(is_gui: bool) -> ! {
 
         WaitForSingleObject(child_handle, INFINITE);
         let mut exit_code = 0u32;
-        check!(GetExitCodeProcess(child_handle, addr_of_mut!(exit_code)));
-        ExitProcess(exit_code);
+        expect_result(
+            GetExitCodeProcess(child_handle, addr_of_mut!(exit_code)),
+            0,
+            || String::from("Error from GetExitCodeProcess"),
+        );
+        exit_with_status(exit_code);
+    }
+}
+
+/// Unwraps the result of the C call by asserting that it doesn't match the `error_code`.
+///
+/// Prints the passed error message if the `actual_result` is equal to `error_code` and exits the process with status 1.
+#[inline]
+fn expect_result<T, F>(actual_result: T, error_code: T, error_message: F) -> T
+where
+    T: Eq,
+    F: FnOnce() -> String,
+{
+    if actual_result == error_code {
+        print_error_and_exit(&error_message());
+    }
+
+    actual_result
+}
+
+#[cold]
+fn print_error_and_exit(message: &str) -> ! {
+    use windows_sys::Win32::{
+        Foundation::*,
+        System::Diagnostics::Debug::{
+            FormatMessageA, FORMAT_MESSAGE_ALLOCATE_BUFFER, FORMAT_MESSAGE_FROM_SYSTEM,
+            FORMAT_MESSAGE_IGNORE_INSERTS,
+        },
+    };
+
+    let err = unsafe { GetLastError() };
+    eprintln!("Received error code: {}", err);
+    let mut msg_ptr: *mut u8 = core::ptr::null_mut();
+    let size = unsafe {
+        FormatMessageA(
+            FORMAT_MESSAGE_ALLOCATE_BUFFER
+                | FORMAT_MESSAGE_FROM_SYSTEM
+                | FORMAT_MESSAGE_IGNORE_INSERTS,
+            null(),
+            err,
+            0,
+            // Weird calling convention: this argument is typed as *mut u16,
+            // but if you pass FORMAT_MESSAGE_ALLOCATE_BUFFER then you have to
+            // *actually* pass in a *mut *mut u16 and just lie about the type.
+            // Getting Rust to do this requires some convincing.
+            core::ptr::addr_of_mut!(msg_ptr) as *mut _ as _,
```

Comment:
Hmmm. Since you're using [`FormatMessageA`](https://docs.rs/windows-sys/latest/windows_sys/Win32/System/Diagnostics/Debug/fn.FormatMessageA.html) and the type of `lpbuffer` is `PSTR = *mut u8`, then I think this comment should say `*mut 8` instead?

In any case, I think most of this strangeness is coming from using `FORMAT_MESSAGE_ALLOCATE_BUFFER`. I'd suggest just putting a buffer on the stack and having this function read into it. And also use `FormatMessageW`. The `A` routines are legacy AFAIK.

You might be able to just copy what std does: https://github.com/rust-lang/rust/blob/bf9229a2e366b4c311f059014a4aa08af16de5d8/library/std/src/sys/windows/os.rs#L26-L80

Reply:
Ahhhh and I just realized you didn't write this code... OK. I'll understand if you want to just keep it the way it is now because it presumably works.